### DISCUSSION TITLE
BMP388 changes for BMP390 using DTS and set Chip-id

### DISCUSSION THREAD

**Discussion by ruehlchris** [zephyrproject-rtos/zephyr#77624] [General]
Hi, with the BMP390 drop in replacement for Bosch BMP388 it would be easy to use the existing driver with a minor change to check the chip-id (0x50=bmp388; 0x60=bmp390).
Changes would be:
dts/bindings bosch,bmp388.yaml add the description for chip-id here.
bmp388.c use data->chip_id from new struct member which read the chip_id from dts defines
bmp388.h remove the obsolete define for the 388 only chip-id.

bosch_bmp390: bmp390@77 {
                compatible = "bosch,bmp388";
                reg = <0x77>;
                odr = "12.5";
                osr-press = <8>;  
                osr-temp = <1>;  
                iir-filter = <3>;
                chip-id = <0x60>;  /* 0x50 = bmp388;  0x60 = bmp390 */
                status="disabled"
  };

On the side, in the bmp388 driver are some improvements can do.
- introduce profiles from Bosch recommended settings (BMP390 datasheet , 3.5. Filter selection) for a proper setting using dts.
- replace bit-shift operation with sys_get_le24()
- alternative conversion based on floating point. (BMP390 datasheet, 8.4. Calibration coefficient)

I have started some work on it for a project we working on replacing bmp180 with bmp390.

Please let me know what you think. 

Cheers.
Chris

**Comment by gherlein** [comment]
@ruehlchris are you working on this?  I am new to Zephyr but I can help test.  I have a BMP390 and very much want to use it in a project.

**Comment by ruehlchris** [comment]
Hi, @gherlein
Yes have already a pull request and need to do some cosmetics only. 
Tests done for I2C and SPI bus. Have them running side to side.
Pull request #77907
